src/js/components/Breadcrumb.js

```diff
@@ -183,86 +173,43 @@ class Breadcrumb extends React.Component {
-    let splitLabelRegex = /[\s-]+/;
-    let joinLabelCharacter = ' ';
-    let params = {};
-
-    if (labelIsParam) {
-      // Extract param value as label e.g. (:userID -> Foo)
-      // and add param to params.
-      //
-      // Remove colon first
-      let paramName = label.slice(1);
-      // Decode value
-      label = decodeURIComponent(routeParams[paramName]);
-      params[paramName] = label;
-    } else {
-      // Split Label on space and - then capitalize each word and join.
-      // /foo-bar/ --> Foo Bar
-      label = label.split(splitLabelRegex).map(function (word) {
-        return StringUtil.capitalize(word);
-      }).join(joinLabelCharacter);
+
+  buildCrumbs(routeName) {
+    let {namedRoutes} = this.context.router;
+    let route = namedRoutes[routeName];
+
+    if (!route || !route.buildBreadCrumb) {
+      return [];
     }
-    // Can return multiple route objects to get rendered in crumbs.
-    // Useful for splitting a route parameter like a file path.
-    return [{label, route, params}];
-  }
 
-  renderCrumbsFromRoute() {
-    let {buildCrumb, shift} = this.props;
+    let crumbConfiguration = route.buildBreadCrumb();
+    let crumbs = crumbConfiguration.getCrumbs(this.context.router);
 
-    if (buildCrumb == null) {
-      buildCrumb = this.buildCrumb;
+    if (crumbConfiguration.parentCrumb) {
+      crumbs = this.buildCrumbs(crumbConfiguration.parentCrumb).concat(crumbs);
     }
 
-    let crumbKey = 1;
-    let namedRoutes = this.getCurrentNamedRoutes();
-    let params = this.getCurrentRouteParams();
-    // Remove n items from beginning
-    let paths = Object.keys(namedRoutes).slice(shift);
-
-    return paths.reduce((crumbs, path, pathIndex) => {
-      let name = namedRoutes[path];
-      let pathParts = path.split('/');
-      // Decode Label
-      let label = decodeURIComponent(pathParts[pathParts.length - 1]);
-      let labelIsParam = label.charAt(0) === ':' && label.slice(1) in params;
-
-      let crumbObjects = buildCrumb(
-        label,
-        name,
-        params,
-        labelIsParam
-      );
+    return crumbs;
+  }
+
+  renderCrumbsFromRoute() {
+    let {router} = this.context;
+    let routes = router.getCurrentRoutes();
+    let currentRoute = routes[routes.length - 1];
+    let crumbs = this.buildCrumbs(currentRoute.name);
 
-      crumbObjects.forEach((crumbParams, crumbIndex) => {
-        let {label, route, params} = crumbParams;
+    crumbs = crumbs.slice(this.props.shift);
 
-        crumbs.push(this.getCrumb(label, route, params, crumbKey++));
+    let crumbKey = 0;
 
-        if (pathIndex !== paths.length - 1
-          || crumbIndex !== crumbObjects.length - 1) {
+    return crumbs.reduce((memo, crumb, crumbIndex) => {
+      memo.push(this.getCrumb(crumb, ++crumbKey));
 
-          crumbs.push(this.getCrumbDivider(crumbKey++));
-        }
-      });
+      if (crumbs.length - 1 !== crumbIndex) {
+        memo.push(this.getCrumbDivider(++crumbKey));
```

Comment:
It saves a variable, which safes memory and lines of code.
Of course it is different. :)